### ISSUE TITLE
JsIgnore on one constructor skips others

### ISSUE THREAD

**Issue by niloc132** [Vertispan/jsinterop-ts-defs#31]
Given a JsType annotated class, with two public constructors, one marked `@JsIgnore` and one not, the non-ignored constructor should be exported, and should be noted in the .d.ts output as the constructor for the type.

```java
@JsType
public class Foo {
  public Foo() {
  }

  @JsIgnore
  public Foo(String str) {
    this();// must always call the real js constructor so that instanceof works, this is a jsinterop restriction
  }
}
```

Expected
```typescript
export class Foo {
  constructor();
}
```

Actual
```typescript
export class Foo {
  protected constructor();
}
```

Note that adding `@JsConstructor` to the non-ignored constructor does not work as a workaround

**Comment by niloc132** [comment]
See JsRollupConfig, JsTreeTableConfig, and JsTotalsTableConfig in my .d.ts branch for deephaven-core.

**Comment by niloc132** [comment]
I believe this was fixed by #33?
